### PR TITLE
Initial GitHub actions

### PR DESCRIPTION
**Pull request by orelmisan**
Add the following job to the GitHub actions workflow:

1. lint.
2. unit-test.
3. build.

All of these jobs should pass in order to merge a PR.

~~This PR depends on PR #2.~~

### CODE REVIEW

**Comment by ormergi** [comment]
There are self-contained jobs for lint, build, and unit-test, is it possible to set a dependency of one on another?
for example: 
- If lint is passing, run build.
- If the build is passing, run unit-test.
- When running unit-tests, use Go build-cache when running tests (instead of fetching all dependencies and compiling again). Maybe with the Github action job spec `needs` keyword.

**Comment by EdDev** [comment]
> There are self-contained jobs for lint, build, and unit-test, is it possible to set a dependency of one on another?

if the cost is low, there is no good reason to invest in it. Having things running independently has its pros, e.g. detecting multiple problems.
The added value for using pipelining presents itself when there is a resource problem and slow jobs.

**Comment by orelmisan** [comment]
> There are self-contained jobs for lint, build, and unit-test, is it possible to set a dependency of one on another? for example:
> 
> * If lint is passing, run build.
> * If the build is passing, run unit-test.
> * When running unit-tests, use Go build-cache when running tests (instead of fetching all dependencies and compiling again). Maybe with the Github action job spec `needs` keyword.

The project is currently very small, so the time it takes to perform each step is quite short.
Another pro for having the jobs independent and parallel for the moment, is the shorter time it takes to complete all jobs in comparison to the time it would take if we serialize them.

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,28 @@
+CONTAINER_ENGINE ?= podman
+
+CHECKUP_IMAGE_NAME := quay.io/kiagnose/kubevirt-rt-checkup
+CHECKUP_IMAGE_TAG ?= devel
+
+GO_IMAGE_NAME := docker.io/library/golang
```
Comment (EdDev) [inline Makefile]: dockerhub is known to limit pulls, this may become a problem later.

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+GO_IMAGE_TAG := 1.19.4-alpine3.17
+
+LINTER_IMAGE_NAME := docker.io/golangci/golangci-lint
```
Comment (EdDev) [inline Makefile]: Same

**Review by EdDev — DISMISSED** [review]
It looks nice, although I am not in favor of using makefiles and container images when not a must.
Taking a tool like makefile and using it for simple scripting does not compute right for me.
The container image to do all targets is also limiting, as the CI already has tools to test on many go versions. In the current state, you will need to update frequently the config to choose the latest version.

But this is for you to choose at the end.
Overall, except of the minor inline comments, this looks good.

File: Makefile (hunk 1 above)
Comment (orelmisan) [reply Makefile]: Yes I am aware of this rate limit.
I have found a mirror in quay, but it is not very up-to-date with the official one:
https://quay.io/repository/projectquay/golang?tab=tags

**Comment by orelmisan** [comment]
Rebase

**Review by EdDev — APPROVED** [review]

**Pull request merged**